examples_tests/23.Autoexposure/main.cpp

```diff
@@ -4,20 +4,771 @@
+				bufferRange.size = sizeof(tonemappingParams);
+				bufferRange.buffer = toneParamsLumaOutputBuffer[i];
+				utilities->updateBufferRangeViaStagingBuffer(
+					fence.get(),
+					queues[CommonAPI::InitOutput::EQT_TRANSFER_UP],
+					bufferRange,
+					&tonemappingParams);
+				logicalDevice->blockForFences(1u, &fence.get());
+			}
+		}
+
+		// create and update tonemapping descriptor sets
+		{
+			const auto pipelineLayout = tonemappingPipeline->getLayout();
+			const auto dsLayout = pipelineLayout->getDescriptorSetLayout(0u);
+
+			const uint32_t setCount = 3u;
+			auto dsPool = logicalDevice->createDescriptorPoolForDSLayouts(video::IDescriptorPool::ECF_NONE, &dsLayout, &dsLayout + 1ull, &setCount);
+
+			for (uint32_t i = 0u; i < SC_IMG_COUNT; ++i)
+			{
+				tonemappingDS[i] = logicalDevice->createGPUDescriptorSet(dsPool.get(), core::smart_refctd_ptr<const video::IGPUDescriptorSetLayout>(dsLayout));
+
+				ToneMapperClass::updateDescriptorSet<MeterMode>(
+					logicalDevice.get(),
+					tonemappingDS[i].get(),
+					tonemapperOutImageViews[i],
+					toneParamsLumaOutputBuffer[i],
+					imageToTonemapView,
+					lumaMeterUbo);
+			}
+		}
+
+		// create luma metering pipeline
+		{
+			auto cpuSpecShader = LumaMeterClass::createShader(assetManager->getGLSLCompiler(), inputColorSpace, MeterMode, MinLuma, MaxLuma);
+			auto gpuUnspecShader = logicalDevice->createGPUShader(core::smart_refctd_ptr<asset::ICPUShader>(cpuSpecShader->getUnspecialized()));
+			auto gpuSpecShader = logicalDevice->createGPUSpecializedShader(gpuUnspecShader.get(), cpuSpecShader->getSpecializationInfo());
+
+			auto pipelineLayout = LumaMeterClass::getDefaultPipelineLayout(logicalDevice.get());
+			lumaMeteringPipeline = logicalDevice->createGPUComputePipeline(nullptr, std::move(pipelineLayout), std::move(gpuSpecShader));
+		}
+
+		// create and update luma metering descriptor sets
+		{
+			auto dsLayout = core::smart_refctd_ptr<const video::IGPUDescriptorSetLayout>(lumaMeteringPipeline->getLayout()->getDescriptorSetLayout(0u));
+			const uint32_t setCount = SC_IMG_COUNT;
+			auto descriptorPool = logicalDevice->createDescriptorPoolForDSLayouts(video::IDescriptorPool::ECF_NONE, &dsLayout.get(), &dsLayout.get() + 1ull, &setCount);
+
+			for (uint32_t i = 0u; i < SC_IMG_COUNT; ++i)
+			{
+				lumaMeteringDS[i] = logicalDevice->createGPUDescriptorSet(descriptorPool.get(), core::smart_refctd_ptr(dsLayout));
+
+				asset::SBufferRange<video::IGPUBuffer> lumaOutputBufferRange = {};
+				lumaOutputBufferRange.offset = sizeof(ToneMapperClass::Params_t<TMO>);
+				lumaOutputBufferRange.size = toneParamsLumaOutputBuffer[i]->getCachedCreationParams().declaredSize - lumaOutputBufferRange.offset;
+				lumaOutputBufferRange.buffer = toneParamsLumaOutputBuffer[i];
+
+				LumaMeterClass::updateDescriptorSet<MeterMode>(
+					lumaMeteringDS[i].get(),
+					lumaMeterUbo,
+					lumaOutputBufferRange,
+					imageToTonemapView,
+					logicalDevice.get());
+			}
+		}
```

Comment:
if we don't need `toneParamsLumaOutputBuffer` in multiple copies, then you don't need this descriptor set in multiple copies either